### PR TITLE
FIM: Follow Symbolic Link test updated

### LINKED ISSUE #502: Update test_follow_symbolic_link
`test_delete_target.py` must be updated to match the new expected behaviour of symbolic link files and targeting directories when those directories are removed. 

Having a Whodata-monitored sybolic link file targeting a non-empty directory here is the new steps the test have to follow:

```
1. Remove the SysLink's target directory and wait for a "deleted" event.
2. Create again the directory.
3. Wait for "Audit rule removed" message.
4. Wait for "Reloading Audit rules" message.
5. Wait for "Reloaded audit rule for monitoring directory: DIRECTORY_NAME" message.
6. Create a file inside the recently-created folder and wait for the "added" event.
7. Wait for "Links check finalized"
8. Wait for "Audit rule loaded: -w ... -p"
9. Modify the new file and wait for the "modify" event.
```
This change has been suggested and approved by Core team.

### PR DESCRIPTION
**Pull request by CarlosRS9**
This closes #502.

This PR updates `test_delete_target.py` to match the new expected behaviour of symbolic link files and targeting directories when those directories are removed. `Fim.py` has a new callback needed for this purpouse.

The test now performs the followings steps:
```
1. Remove the SysLink's target directory and wait for a "deleted" event.
2. Create again the directory.
3. Wait for "Audit rule removed" message.
4. Wait for "Reloading Audit rules" message.
5. Wait for "Reloaded audit rule for monitoring directory: DIRECTORY_NAME" message.
6. Create a file inside the recently-created folder and wait for the "added" event.
7. Wait for "Links check finalized"
8. Wait for "Audit rule loaded: -w ... -p"
9. Modify the new file and wait for the "modify" event.
This change has been suggested and approved by Core team.
```

## Test results

```
============================= test session starts ==============================
platform linux -- Python 3.6.8, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /vagrant/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.8.0, html-2.0.1
collected 69 items                                                             

test_fim/test_follow_symbolic_link/test_change_target.py .ss..ss..ss.    [ 17%]
test_fim/test_follow_symbolic_link/test_change_target_inside_folder.py . [ 18%]
ss..ss..ss.                                                              [ 34%]
test_fim/test_follow_symbolic_link/test_delete_symlink.py .ss..ss..ss.   [ 52%]
test_fim/test_follow_symbolic_link/test_delete_target.py .ss..ss..ss.    [ 69%]
test_fim/test_follow_symbolic_link/test_monitor_symlink.py .ss..ss..ss.  [ 86%]
test_fim/test_follow_symbolic_link/test_not_following_symbolic_link.py . [ 88%]
.....                                                                    [ 95%]
test_fim/test_follow_symbolic_link/test_revert_symlink.py ...            [100%]

=========== 39 passed, 30 skipped, 2 warnings in 1496.92s (0:24:56) ============


```

### CODE REVIEW

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py [hunk 1]
```diff
@@ -2,15 +2,17 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
+import os
 import pytest
+import time
```
Comment (vicferpoy) [inline tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Import not used.

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
-    wait_for_symlink_check, wait_for_audit, testdir_target, testdir_not_target, delete_f
+    wait_for_symlink_check, wait_for_audit, testdir_target, testdir_not_target, delete_f, symlink_interval
 # noinspection PyUnresolvedReferences
 from test_fim.test_follow_symbolic_link.common import test_directories, extra_configuration_before_yield, \
     extra_configuration_after_yield
 
-from wazuh_testing.fim import (generate_params, create_file, REGULAR, callback_detect_event,
-                               check_time_travel, modify_file_content, LOG_FILE_PATH)
+from wazuh_testing.fim import (generate_params, create_file, REGULAR, callback_detect_event, callback_audit_removed_rule,
+                               callback_audit_reloaded_rule, callback_audit_reloading_rules, check_time_travel, modify_file_content, LOG_FILE_PATH)
```
Comment (vicferpoy) [inline tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Check PEP8

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py [hunk 3]
```diff
@@ -76,27 +78,46 @@ def test_symbolic_delete_target(tags_to_apply, main_folder, aux_folder, get_conf
...
     assert 'deleted' in delete['data']['type'] and file1 in delete['data']['path'], \
         f"'deleted' event not matching for {file1}"
 
-    # If syscheck is monitoring with whodata, wait for audit to reload rules
-    wait_for_audit(whodata, wazuh_log_monitor)
-    wait_for_symlink_check(wazuh_log_monitor)
+    if tags_to_apply == {'monitored_dir'} and whodata:
+        os.makedirs(main_folder, exist_ok=True, mode=0o777)
+        wazuh_log_monitor.start(timeout=3, callback=callback_audit_removed_rule,
+                                error_message='Did not receive expected "Monitored directory \'{main_folder}\' was removed: Audit rule removed')
```
Comment (vicferpoy) [inline tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Check PEP8

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py [hunk 4]
```diff
@@ -76,27 +78,46 @@ def test_symbolic_delete_target(tags_to_apply, main_folder, aux_folder, get_conf
...
+        wazuh_log_monitor.start(timeout=symlink_interval, callback=callback_audit_reloading_rules,
+                                error_message='Did not receive expected "Reloading Audit rules" event')
+        wazuh_log_monitor.start(timeout=symlink_interval, callback=callback_audit_reloaded_rule,
+                                error_message='Did not receive expected "Reloaded audit rule for monitoring directory: \'{main_folder}\'" event')
```
Comment (vicferpoy) [inline tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Check PEP8

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py (hunk 1 above)
Comment (CarlosRS9) [reply tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Good catch. Fixed.

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py (hunk 2 above)
Comment (CarlosRS9) [reply tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Done

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py (hunk 3 above)
Comment (CarlosRS9) [reply tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Done

File: tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py (hunk 4 above)
Comment (CarlosRS9) [reply tests/integration/test_fim/test_follow_symbolic_link/test_delete_target.py]: Done

File: deps/wazuh_testing/wazuh_testing/fim.py [hunk 5]
```diff
@@ -729,6 +729,13 @@ def callback_audit_event_too_long(line):
     return None
 
 
+def callback_audit_reloading_rules(line):
+    match = re.match(r'.*Reloading Audit rules', line)
+    if match:
+        return True
+    return None
```
Comment (crd1985) [inline deps/wazuh_testing/wazuh_testing/fim.py]:
```suggestion
```

**Pull request merged**